### PR TITLE
IMPROVE: sync resources during Subscriptions Update process

### PR DESCRIPTION
**Pull request by merefield**
- makes the product id migration more robust
- updates product information when user runs subscriptions tab update process

### CODE REVIEW

File: lib/subscription_client/resources.rb [hunk 1]
```diff
@@ -34,7 +34,13 @@ def setup_resources
   end
 
   def find_resources
-    find_plugins
+    resources = find_plugins
```
Comment (angusmcleod) [inline lib/subscription_client/resources.rb:37]: @merefield Could you explain the thinking behind this change? Looks like it's not changing the functionality and will break the use of TEST_SUBSCRIPTION_URL.

Comment (merefield) [reply lib/subscription_client/resources.rb:37]: Yes, without refactoring here and adding a layer, you can't stub the method acts as the directory search.

Comment (angusmcleod) [reply lib/subscription_client/resources.rb:37]: Ah I see, that makes sense. Don't we want to keep TEST_SUBSCRIPTION_URL functionality too though?

**Comment by merefield** [comment]
@angusmcleod please hold of from further review, I've seen an issue and I'm going to resolve it.

**Comment by merefield** [comment]
@angusmcleod apologies for the disruption, please re-review.  I've now confirmed that on a standard install, the update button on the subscriptions page resolves the nil products issue.

File: lib/subscription_client/resources.rb (hunk 1 above)
Comment (merefield) [reply lib/subscription_client/resources.rb:37]: TEST_SUBSCRIPTION_URL is still respected.  I've just checked that on a standard install connecting to `test2`

**Pull request merged**